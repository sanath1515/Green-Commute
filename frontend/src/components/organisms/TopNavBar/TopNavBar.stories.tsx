import React from "react";
import TopNavBar from "./TopNavBar";

export default {
  title: "Organisms/TopNavBar",
  component: TopNavBar,
};

const Template = (args) => <TopNavBar {...args} />;

export const Primary = Template.bind({});

Primary.args = {
  location: "East Mareedpally, Secunderabad",
  src:
    "https://cdn.zeplin.io/5fa28fcbff79c58958e4e1a6/assets/21469ead-aeb7-47f1-900e-6a85e9034b59-3x.png",
};
