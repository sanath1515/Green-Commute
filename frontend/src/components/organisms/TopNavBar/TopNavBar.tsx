import * as React from "react";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import { Typography } from "@material-ui/core";
import { COMMUTE_ROUTES, NAV_ICONS } from "../../../Constants";
import { Avatars } from "../../atoms/Avatar/Avatar";
import { Navicons } from "../../atoms/dashboardicons/Navicons";
import { makeStyles } from "@material-ui/core/styles";
import KeyboardArrowDownIcon from "@material-ui/icons/KeyboardArrowDown";

const useStyles = makeStyles((theme) => ({
  root: {
    width: 1180,
    height: "80px",
    boxShadow: "none",
    paddingTop: "8px",
    paddingBottom: "26px",
    background: "white",
    border: "solid 1px #e3f3f6",
    paddingLeft: "60px",
    paddingRight: "40px",
    marginTop: theme.spacing(-4.5),
    marginLeft: theme.spacing(-2),
  },
  icon: {
    width: "24px",
    height: "24px",
    paddingTop: "33px",
    paddingBottom: "28px",
    paddingRight: "28px",
    color: "#9bbdcb",
    paddingLeft: "10px",
  },
  location: {
    width: "265px",
    height: "24px",
    flexGrow: 1,
    color: "#19293b",
    marginLeft: "-18px",
    paddingTop: "1px",
  },
  user: {
    width: "87px",
    height: "18px",
    color: "#19293b",
    paddingLeft: "10px",
  },
  down: {
    color: "black",
    width: "20px",
    height: "20px",
    flexGrow: 0,
    objectFit: "contain",
    marginLeft: "5px",
  },
}));

type navProps = {
  location?: string;
  src?: string;
};
const TopNavBar = ({ location, src }: navProps) => {
  const classes = useStyles();
  return (
    <>
      <AppBar position="static" className={classes.root}>
        <Toolbar>
          <Navicons name={NAV_ICONS[7]} className={classes.icon} />
          <Typography variant="subtitle1" className={classes.location}>
            {location}
          </Typography>
          <Navicons name={NAV_ICONS[0]} className={classes.icon} />
          <Navicons name={NAV_ICONS[1]} className={classes.icon} />
          <Avatars src={src} />
          <Typography className={classes.user} variant="subtitle2">
            {COMMUTE_ROUTES.userName}
          </Typography>
          <KeyboardArrowDownIcon />
        </Toolbar>
      </AppBar>
    </>
  );
};

export default TopNavBar;
