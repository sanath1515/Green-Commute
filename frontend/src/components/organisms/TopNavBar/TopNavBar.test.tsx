/**
 * @jest-environment jsdom
 */
import { fireEvent, render, screen } from "@testing-library/react";
import TopNavBar from "./TopNavBar";

describe("TopNavBar Component", () => {
  test("Matches Snapshot", () => {
    const { asFragment } = render(<TopNavBar />);
    expect(asFragment.name).toMatchSnapshot();
  });
});
