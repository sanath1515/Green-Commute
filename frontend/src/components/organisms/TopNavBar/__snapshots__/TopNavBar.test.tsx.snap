// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopNavBar Component Matches Snapshot 1`] = `"asFragment"`;
